rules/otbn.bzl

```diff
@@ -175,3 +214,21 @@ otbn_binary = rv_rule(
     toolchains = ["@rules_cc//cc:toolchain_type"],
     incompatible_use_toolchain_transition = True,
 )
+
+otbn_test = rv_rule(
+    implementation = _otbn_test,
+    test = True,
+    attrs = {
+        "srcs": attr.label_list(allow_files = True),
+        "deps": attr.label_list(providers = [DefaultInfo]),
+        "_cc_toolchain": attr.label(default = Label("@bazel_tools//tools/cpp:current_cc_toolchain")),
+        "_otbn_as": attr.label(default = "//hw/ip/otbn/util:otbn-as", allow_single_file = True),
+        "_otbn_ld": attr.label(default = "//hw/ip/otbn/util:otbn-ld", allow_single_file = True),
+        "_otbn_data": attr.label(default = "//hw/ip/otbn/data:all_files", allow_files = True),
+        "_simulator": attr.label(default = "//hw/ip/otbn/dv/otbnsim:standalone.py", allow_single_file = True),
```

Comment:
Doesn't this need to depend on the rest of the stuff in `otbnsim`? Or does Bazel's Python tracking do some magic to follow dependencies?

Reply:
It seems to do some magic; I tested it locally and it does run even though it copies the file to a temporary directory.

Reply:
I have no idea why this works... but it absolutely should not. Bazel does not look at source files to determine dependencies, so I suspect it only _happens_ to work due to Python being extremely hard to make hermetic.

Reply:
This would need to become a `py_binary` rule for bazel to track the python deps (as you hint in your TODO comment). Right now, bazel just thinks it's a file

Reply:
> Can you list the dependencies you expect it to need? If it's doing magic, let's try not to rely on it.

It's the standalone endpoint for the simulator, so I expect it'll depend on the 19 files under `hw/ip/otbn/dv/otbnsim/sim/` as well as a few additional dependencies from `hw/ip/otbn/util/shared/` and `util/reggen/`. It'll also need the external dependency `pyelftools`. So it'll be quite a long chain of dependencies to bazelify, which is why I was thinking it would make sense as a separate change.